pkg/apis/componentconfig/types.go

```diff
@@ -62,3 +62,213 @@ const (
 	ProxyModeUserspace ProxyMode = "userspace"
 	ProxyModeIPTables  ProxyMode = "iptables"
 )
+
+type KubeletConfiguration struct {
```

Comment:
per-Brian's comment, a few of these have empty defaults.  Is it possible to add 'omitempty' to the json tag for those?